data/state/peerAccount.go

```diff
@@ -0,0 +1,369 @@
+func (a *PeerAccount) SetBLSPublicKeyWithJournal(pubKey []byte) error {
+	if len(pubKey) < 1 {
+		return ErrNilBLSPublicKey
+	}
+
+	entry, err := NewPeerJournalEntryBLSPublicKey(a, a.BLSPublicKey)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.BLSPublicKey = pubKey
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetStakeWithJournal sets the account's stake, saving the old stake before changing
+func (a *PeerAccount) SetStakeWithJournal(stake *big.Int) error {
+	if stake == nil {
+		return ErrNilStake
+	}
+
+	entry, err := NewPeerJournalEntryStake(a, a.Stake)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.Stake = stake
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetJailTimeWithJournal sets the account's jail time, saving the old state before changing
+func (a *PeerAccount) SetJailTimeWithJournal(jailTime TimePeriod) error {
+	entry, err := NewPeerJournalEntryJailTime(a, a.JailTime)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.JailTime = jailTime
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetCurrentShardIdWithJournal sets the account's shard id, saving the old state before changing
+func (a *PeerAccount) SetCurrentShardIdWithJournal(shId uint32) error {
+	entry, err := NewPeerJournalEntryCurrentShardId(a, a.CurrentShardId)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.CurrentShardId = shId
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetNextShardIdWithJournal sets the account's shard id, saving the old state before changing
+func (a *PeerAccount) SetNextShardIdWithJournal(shId uint32) error {
+	entry, err := NewPeerJournalEntryNextShardId(a, a.NextShardId)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.NextShardId = shId
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetNodeInWaitingListWithJournal sets the account's nodes status whether in waiting list, saving the old state before
+func (a *PeerAccount) SetNodeInWaitingListWithJournal(nodeInWaitingList bool) error {
+	entry, err := NewPeerJournalEntryInWaitingList(a, a.NodeInWaitingList)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.NodeInWaitingList = nodeInWaitingList
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// IncreaseValidatorSuccessRateWithJournal increases the account's number of successful signing,
+// saving the old state before changing
+func (a *PeerAccount) IncreaseValidatorSuccessRateWithJournal() error {
+	entry, err := NewPeerJournalEntryValidatorSuccessRate(a, a.ValidatorSuccessRate)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.ValidatorSuccessRate.NrSuccess++
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// DecreaseValidatorSuccessRateWithJournal increases the account's number of missed signing,
+// saving the old state before changing
+func (a *PeerAccount) DecreaseValidatorSuccessRateWithJournal() error {
```

Comment:
IncreaseValidatorFailureRateWithJournal instead DecreaseValidatorSuccessRateWithJournal ?

Reply:
the fields name is successRate, I wanted to mask what happens in the back.